### PR TITLE
Adding Changelog for v0.5.1

### PR DESCRIPTION
**Pull request by robscott**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
This adds the changelog for v0.5.1. This also bumps version numbers from v0.5.0 to v0.5.1 where applicable. Approving this PR means that you're approving a release of v0.5.1 based on this commit.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/hold for consensus

### CODE REVIEW

**Review by shaneutt — APPROVED** [review]

**Comment by shaneutt** [comment]
I'll leave the `lgtm` for one other person just since this will result in a release, but it does LGTM.

**Comment by robscott** [comment]
Last push updated changelog to include #1382 from @skriss. I think this is good to go now. Hopefully can get this release out tomorrow.

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +18,39 @@
...
+* The spec has been clarified to state that "Accepted" should be used instead of
+  "Attached" on HTTPRoute.
+  [#1382](https://github.com/kubernetes-sigs/gateway-api/pull/1382)
+
+### Webhook:
+
+* The duplicate gateway-system namespace definitions have been removed.
+  [#1387](https://github.com/kubernetes-sigs/gateway-api/pull/1387)
+* The webhook has been updated to watch v1beta1.
+  [1365](https://github.com/kubernetes-sigs/gateway-api/pull/1368)
```
Comment (gyohuangxin) [inline CHANGELOG.md:41]: "#" is missed before issue number?

File: CHANGELOG.md [hunk 2]
```diff
@@ -17,6 +18,39 @@
...
+
+### Conformance:
+
+* The expected condition for a cross-namespace certificate reference that has
+  not been allowed by a ReferenceGrant has been changed from
+  "InvalidCertificateRef" to "RefNotPermitted" to more closely match the spec.
+  [#1351](https://github.com/kubernetes-sigs/gateway-api/pull/1351)
+* A new test has been added to cover when a Gateway references a Secret that
+  does not exist
+  [1334](https://github.com/kubernetes-sigs/gateway-api/pull/1334)
```
Comment (gyohuangxin) [inline CHANGELOG.md:51]: Ditto

**Comment by youngnick** [comment]
LGTM with the changes notes by @gyohuangxin.

**Pull request merged**